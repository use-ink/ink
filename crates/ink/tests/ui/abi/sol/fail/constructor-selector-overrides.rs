#![allow(unexpected_cfgs)]

#[ink::contract]
mod contract {
    #[ink(storage)]
    pub struct Contract {}

    impl Contract {
        #[ink(constructor, selector = 1)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
