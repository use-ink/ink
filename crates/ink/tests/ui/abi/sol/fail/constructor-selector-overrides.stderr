error: constructor `selector` attributes are not supported in Solidity ABI compatibility mode
 --> tests/ui/abi/sol/fail/constructor-selector-overrides.rs:9:28
  |
9 |         #[ink(constructor, selector = 1)]
  |                            ^^^^^^^^^^^^
